src/OpenTelemetry.AutoInstrumentation/Diagnostics/SdkSelfDiagnosticsEventListener.cs

```diff
@@ -0,0 +1,111 @@
+// <copyright file="SdkSelfDiagnosticsEventListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+// Source originated from https://github.com/open-telemetry/opentelemetry-dotnet/blob/23609730ddd73c860553de847e67c9b2226cff94/src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace OpenTelemetry.AutoInstrumentation.Diagnostics;
+
+/// <summary>
+/// SdkSelfDiagnosticsEventListener class enables the events from OpenTelemetry event sources
+/// and write the events to the OpenTelemetry.AutoInstrumentation logger
+/// </summary>
+internal class SdkSelfDiagnosticsEventListener : EventListener
+{
+    private const string EventSourceNamePrefix = "OpenTelemetry-";
```

Comment:
This will collect EventSource from OpenTelemetry .NET SDK and Auto Instrumentation. SDK already has an in-built EventListener. Do we want to collect events from both SDK and auto-instrumentation.?

Ref: https://github.com/open-telemetry/opentelemetry-dotnet/blob/7ed97be90841221ec2451d8d14a622768cb214b5/src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs#L405

Reply:
I propose that yes we collect events from both SDK and auto-instrumentation. The EventListener you linked is only included in Debug builds of the SDK (which we likely won't be using at all) and to set up the SDK's [SelfDiagnostics](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/README.md#self-diagnostics) EventListener, we would need to place a `OTEL_DIAGNOSTICS.json` file in the working directory which doesn't seem feasible for us.